Updates to Readme

loricooperhdf opened the pull request:
Cleared out some old info (Autotools), added info about priority support/consulting, and made some minor changes so that 2.0 doesn't seem so far in the future. 
<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Removed outdated Autotools info, added priority support details, and updated text for HDF5 2.0 in `README.md`.
> 
>   - **Removed Information**:
>     - Removed outdated Autotools information and references to its removal.
>   - **Support Information**:
>     - Added details about priority support and consulting services in the `HELP AND SUPPORT` section.
>   - **Minor Text Adjustments**:
>     - Updated text to make HDF5 2.0 release seem more imminent.
>     - Corrected link to HDF5 2.0 planning wiki in `RELEASE SCHEDULE`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=HDFGroup%2Fhdf5&utm_source=github&utm_medium=referral)<sup> for 0a08f5d7564bcc176a319a5aabb8f1e670c112a4. You can [customize](https://app.ellipsis.dev/HDFGroup/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

loricooperhdf committed:
Update README.md

Removed mention of autotools removal, removed some other 2.0 is coming stuff. Updated the Help and support section to mention priority support and made some little text changes re the forum.

loricooperhdf committed:
Change URL for planned 2.0 features

loricooperhdf committed:
added forum link under help and support

loricooperhdf committed:
Changed out link for planned 2.0 features

(Matches Binh-Minh's earlier PR.)

byrnHDF commented on README.md:
```diff
@@ -1,16 +1,3 @@
-> [!NOTE]  
```
I think lines 1,2,3 and 9 should be kept. Line for those coming from 1.14.6

loricooperhdf replied:
OK but let's not leave the 2.0 part in when we release 2.0? :)

byrnHDF replied:
yes, once 2.0 is released there will be no need for line 9 anymore.

byrnHDF replied:
Actually, matbe change the
`HDF5 2.0, scheduled for release in Fall 2025, will *only* support the CMake build system.`
to
`Starting with HDF5 2.0, *only* the CMake build system is supported.`

loricooperhdf replied:
Added back. The sidebar of color is broken when I look in preview. Ugh.

loricooperhdf committed:
Added back warning text

(Please confirm the sidebar color looks right, it's not looking right in preview.)

loricooperhdf committed:
Fixed an em-dash.

byrnHDF commented on README.md:
```diff
@@ -1,15 +1,9 @@
 > **Heads Up: HDF5 Dropped Autotools March 10th**
 >
-> The day has passed: the day we've all been dreading—or eagerly anticipating, depending on your perspective. Yes, we have switched to CMake-only builds in HDF5.
+> It's happened—the day we've all been dreading—or eagerly anticipating, depending on your perspective. Yes, we have switched to CMake-only builds in HDF5.
```
This one line (It's happened) I can go with it in or out.

loricooperhdf replied:
Keeping because it's cheeky and we need to wrap this up.

lrknox commented on README.md:
```diff
@@ -1,15 +1,9 @@
-> [!NOTE]  
-> HDF5 version 2.0.0 currently under development
```
We should probably keep the version line at the top of this file, as the version has been posted at the top of the README file since version 1.0 in 1998.  Removing it will make it more difficult to check the version of the code, and will also require updates to bin/h5vers and bin/release.

loricooperhdf replied:
Added back.

glennsong09 reviewed (dismissed):
I think once Larry's suggested change has been made, this looks good.

loricooperhdf committed:
Added back the Note about current version

lrknox commented on README.md:
```diff
@@ -1,15 +1,12 @@
-> [!NOTE]  
+> [!NOTE]
 > HDF5 version 2.0.0 currently under development
```
bin/h5vers and bin/release scripts expect the line with the version to be the first line in the file.  Please remove the [!NOTE} or tell me that it's critical and the scripts should be altered.

loricooperhdf replied:
Ok. For the record, I just put it back the way it was. 

Is it OK that it says currently under development on the same line?

byrnHDF replied:
Yes, the release process will change it to the correct string

loricooperhdf committed:
Removed note so version is on the top line
